public-ui/src/app/data-browser/components/survey-version-table/survey-version-table-react.component.tsx

```diff
@@ -0,0 +1,88 @@
+import {
+    AfterViewInit,
+    Component,
+    ElementRef,
+    EventEmitter,
+    Input,
+    Output,
+    ViewChild,
+    ViewEncapsulation
+} from '@angular/core';
+import * as React from 'react';
+import { FunctionComponent } from 'react';
+import * as ReactDOM from 'react-dom';
+import { BaseReactWrapper } from '../../../data-browser/base-react/base-react.wrapper';
+import { ClrIcon } from '../../../utils/clr-icon';
+
+const containerElementName = 'root';
+
+interface Props {
+    surveys: any;
+}
+
+export class SurveyVersionTableReactComponent extends React.Component<Props, {}> {
+constructor(props: Props) {
+    super(props);
+    this.processSurveys();
+}
+
+processSurveys() {
+    this.props.surveys.sort((a1, a2) => {
+        if (a1.monthNum.split('/')[0] < a2.monthNum.split('/')[0]) {
+            return -1;
+        }
+        if (a1.monthNum.split('/')[0] > a2.monthNum.split('/')[0]) {
+            return 1;
+        }
+        return 0;
+    });
+this.props.surveys.forEach((survey, i) => {
+    survey['pdfLink'] = '/assets/surveys/' + survey.monthName.replace('/', '_') + '_COPE_COVID_English_Explorer.pdf';
+});
```

Comment:
While this works, it is not recommended to modify a component's own props (see [React docs](https://reactjs.org/docs/components-and-props.html#props-are-read-only)). In this case, I think the best way is to copy the original array to a const, then store it in state after modifying:
```suggestion
    const sortedSurveys = [...this.props.surveys].sort((a1, a2) => {
        if (a1.monthNum.split('/')[0] < a2.monthNum.split('/')[0]) {
            return -1;
        }
        if (a1.monthNum.split('/')[0] > a2.monthNum.split('/')[0]) {
            return 1;
        }
        return 0;
    });
sortedSurveys.forEach((survey, i) => {
    survey['pdfLink'] = '/assets/surveys/' + survey.monthName.replace('/', '_') + '_COPE_COVID_English_Explorer.pdf';
});
this.setState({surveys: sortedSurveys});
```

Reply:
Thanks good to know the practice.